instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/internal/semconv/event_monitor.go

```diff
@@ -0,0 +1,166 @@
+
+import (
+	"net"
+	"os"
+	"strconv"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+
+	"go.opentelemetry.io/otel/attribute"
+
+	semconv1210 "go.opentelemetry.io/otel/semconv/v1.21.0"
+	semconv1260 "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// Constants for environment variable keys and versions.
+const (
+	semconvOptIn     = "OTEL_SEMCONV_STABILITY_OPT_IN"
+	semconvOptInDup  = "mongo/dup"
+	semconvOptIn1260 = "mongo"
+)
+
+// EventMonitor is responsible for monitoring events with a specified semantic
+// version.
+type EventMonitor struct {
+	version string
+}
+
+// NewEventMonitor creates an EventMonitor with the version set based on the
+// OTEL_SEMCONV_STABILITY_OPT_IN environment variable.
+func NewEventMonitor() EventMonitor {
+	return EventMonitor{
+		version: strings.ToLower(os.Getenv(semconvOptIn)),
+	}
+}
+
+// AttributeOptions represents options for tracing attributes.
+type AttributeOptions struct {
+	collectionName           string
+	commandAttributeDisabled bool
+}
+
+// AttributeOption is a function type that modifies AttributeOptions.
+type AttributeOption func(*AttributeOptions)
+
+// WithCollectionName is a functional option to set the collection name in
+// AttributeOptions.
+func WithCollectionName(collName string) AttributeOption {
+	return func(opts *AttributeOptions) {
+		opts.collectionName = collName
+	}
+}
+
+// WithCommandAttributeDisabled is a functional option to enable or disable
+// command attributes.
+func WithCommandAttributeDisabled(disabled bool) AttributeOption {
+	return func(opts *AttributeOptions) {
+		opts.commandAttributeDisabled = disabled
+	}
+}
+
+// CommandStartedTraceAttrs generates trace attributes for a CommandStartedEvent
+// based on the EventMonitor version.
+func (m EventMonitor) CommandStartedTraceAttrs(
+	evt *event.CommandStartedEvent,
+	opts ...AttributeOption,
+) []attribute.KeyValue {
+	switch m.version {
+	case semconvOptIn1260:
+		return commandStartedTraceAttrsV1260(evt, opts...)
+	case semconvOptInDup:
+		return append(commandStartedTraceAttrsV1260(evt, opts...), commandStartedTraceAttrsV1210(evt, opts...)...)
+	default:
+		return commandStartedTraceAttrsV1210(evt, opts...)
+	}
+}
+
+// peerInfo extracts the hostname and port from a CommandStartedEvent.
+func peerInfo(evt *event.CommandStartedEvent) (hostname string, port int) {
+	hostname = evt.ConnectionID
+	port = 27017 // Default MongoDB port
+
+	host, portStr, err := net.SplitHostPort(hostname)
+	if err != nil {
+		// If there's an error (likely because there's no port), assume default port
+		// and use ConnectionID as hostname
+		return hostname, port
+	}
+
+	if parsedPort, err := strconv.Atoi(portStr); err == nil {
+		port = parsedPort
+	}
+
+	return host, port
+}
+
+// sanitizeCommand converts a BSON command to a sanitized JSON string.
+// TODO: Sanitize values where possible and limit maximum size.
```

Comment:
Having this as two different TODOs made it clearer that it's two different things that will be tackled separately.

Reply:
Ping about this comment?

```suggestion
// TODO: Sanitize values where possible.
// TODO: Limit maximum size.
```